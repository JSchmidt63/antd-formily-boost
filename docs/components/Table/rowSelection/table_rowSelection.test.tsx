import Checkbox from './checkbox';
import Radio from './radio';
import Hidden from './hidden';

import { render } from '@testing-library/react';
import React from 'react';

beforeAll(() => {
    Object.defineProperty(window, 'matchMedia', {
        value: jest.fn(() => {
            return {
                matches: true,
                addListener: jest.fn(),
                removeListener: jest.fn(),
            };
        }),
    });
});

test('Checkbox', () => {
    const { asFragment } = render(<Checkbox />);
    expect(asFragment()).toMatchSnapshot();
});

test('Radio', () => {
    const { asFragment } = render(<Radio />);
    expect(asFragment()).toMatchSnapshot();
});

test('Hidden', () => {
    const { asFragment } = render(<Hidden />);
    expect(asFragment()).toMatchSnapshot();
});
