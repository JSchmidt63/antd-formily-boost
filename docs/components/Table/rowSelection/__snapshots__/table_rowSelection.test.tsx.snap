// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox 1`] = `
<DocumentFragment>
  <div
    class="ant-space ant-space-vertical"
    style="display: flex;"
  >
    <div
      class="ant-space-item"
      style="margin-bottom: 10px;"
    >
      <div
        class="ant-formily-layout"
      >
        <form>
          <div
            class="ant-table-wrapper antd_formily_table_boost undefined"
          >
            <div
              class="ant-spin-nested-loading"
            >
              <div
                class="ant-spin-container"
              >
                <div
                  class="ant-table"
                >
                  <div
                    class="ant-table-container"
                  >
                    <div
                      class="ant-table-content"
                    >
                      <table
                        style="table-layout: auto;"
                      >
                        <colgroup>
                          <col
                            class="ant-table-selection-col"
                            style="width: 100px;"
                          />
                        </colgroup>
                        <thead
                          class="ant-table-thead"
                        >
                          <tr>
                            <th
                              class="ant-table-cell ant-table-selection-column"
                            >
                              <div
                                class="ant-table-selection"
                              >
                                <label
                                  class="ant-checkbox-wrapper"
                                >
                                  <span
                                    class="ant-checkbox"
                                  >
                                    <input
                                      class="ant-checkbox-input"
                                      type="checkbox"
                                      value=""
                                    />
                                    <span
                                      class="ant-checkbox-inner"
                                    />
                                  </span>
                                </label>
                                <div
                                  class="ant-table-selection-extra"
                                >
                                  <span
                                    class="ant-dropdown-trigger"
                                  >
                                    <span
                                      aria-label="down"
                                      class="anticon anticon-down"
                                      role="img"
                                    >
                                      <svg
                                        aria-hidden="true"
                                        data-icon="down"
                                        fill="currentColor"
                                        focusable="false"
                                        height="1em"
                                        viewBox="64 64 896 896"
                                        width="1em"
                                      >
                                        <path
                                          d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                                        />
                                      </svg>
                                    </span>
                                  </span>
                                </div>
                              </div>
                            </th>
                            <th
                              class="ant-table-cell"
                            >
                              名字
                            </th>
                            <th
                              class="ant-table-cell"
                            >
                              年龄
                            </th>
                          </tr>
                        </thead>
                        <tbody
                          class="ant-table-tbody"
                        >
                          <tr
                            class="ant-table-row ant-table-row-level-0"
                            data-row-key="0"
                          >
                            <td
                              class="ant-table-cell ant-table-selection-column"
                            >
                              <label
                                class="ant-checkbox-wrapper"
                              >
                                <span
                                  class="ant-checkbox"
                                >
                                  <input
                                    class="ant-checkbox-input"
                                    type="checkbox"
                                    value=""
                                  />
                                  <span
                                    class="ant-checkbox-inner"
                                  />
                                </span>
                              </label>
                            </td>
                            <td
                              class="ant-table-cell"
                            >
                              <div
                                class="ant-formily-item ant-formily-item-layout-horizontal ant-formily-item-feedback-layout-terse ant-formily-item-label-align-right ant-formily-item-control-align-left"
                                data-span="1"
                              >
                                <div
                                  class="ant-formily-item-control"
                                >
                                  <div
                                    class="ant-formily-item-control-content"
                                  >
                                    <div
                                      class="ant-formily-item-control-content-component"
                                    >
                                      <span
                                        class="ant-input-affix-wrapper"
                                      >
                                        <input
                                          class="ant-input"
                                          type="text"
                                          value="fish"
                                        />
                                        <span
                                          class="ant-input-suffix"
                                        >
                                          <span />
                                        </span>
                                      </span>
                                    </div>
                                  </div>
                                </div>
                              </div>
                            </td>
                            <td
                              class="ant-table-cell"
                            >
                              123
                            </td>
                          </tr>
                          <tr
                            class="ant-table-row ant-table-row-level-0"
                            data-row-key="1"
                          >
                            <td
                              class="ant-table-cell ant-table-selection-column"
                            >
                              <label
                                class="ant-checkbox-wrapper"
                              >
                                <span
                                  class="ant-checkbox"
                                >
                                  <input
                                    class="ant-checkbox-input"
                                    type="checkbox"
                                    value=""
                                  />
                                  <span
                                    class="ant-checkbox-inner"
                                  />
                                </span>
                              </label>
                            </td>
                            <td
                              class="ant-table-cell"
                            >
                              <div
                                class="ant-formily-item ant-formily-item-layout-horizontal ant-formily-item-feedback-layout-terse ant-formily-item-label-align-right ant-formily-item-control-align-left"
                                data-span="1"
                              >
                                <div
                                  class="ant-formily-item-control"
                                >
                                  <div
                                    class="ant-formily-item-control-content"
                                  >
                                    <div
                                      class="ant-formily-item-control-content-component"
                                    >
                                      <span
                                        class="ant-input-affix-wrapper"
                                      >
                                        <input
                                          class="ant-input"
                                          type="text"
                                          value="cat"
                                        />
                                        <span
                                          class="ant-input-suffix"
                                        >
                                          <span />
                                        </span>
                                      </span>
                                    </div>
                                  </div>
                                </div>
                              </div>
                            </td>
                            <td
                              class="ant-table-cell"
                            >
                              456
                            </td>
                          </tr>
                          <tr
                            class="ant-table-row ant-table-row-level-0"
                            data-row-key="2"
                          >
                            <td
                              class="ant-table-cell ant-table-selection-column"
                            >
                              <label
                                class="ant-checkbox-wrapper"
                              >
                                <span
                                  class="ant-checkbox"
                                >
                                  <input
                                    class="ant-checkbox-input"
                                    type="checkbox"
                                    value=""
                                  />
                                  <span
                                    class="ant-checkbox-inner"
                                  />
                                </span>
                              </label>
                            </td>
                            <td
                              class="ant-table-cell"
                            >
                              <div
                                class="ant-formily-item ant-formily-item-layout-horizontal ant-formily-item-feedback-layout-terse ant-formily-item-label-align-right ant-formily-item-control-align-left"
                                data-span="1"
                              >
                                <div
                                  class="ant-formily-item-control"
                                >
                                  <div
                                    class="ant-formily-item-control-content"
                                  >
                                    <div
                                      class="ant-formily-item-control-content-component"
                                    >
                                      <span
                                        class="ant-input-affix-wrapper"
                                      >
                                        <input
                                          class="ant-input"
                                          type="text"
                                          value="dog"
                                        />
                                        <span
                                          class="ant-input-suffix"
                                        >
                                          <span />
                                        </span>
                                      </span>
                                    </div>
                                  </div>
                                </div>
                              </div>
                            </td>
                            <td
                              class="ant-table-cell"
                            >
                              789
                            </td>
                          </tr>
                        </tbody>
                      </table>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div>
            {"data":[{"name":"fish","age":123,"_checked":false},{"name":"cat","age":456,"_checked":false},{"name":"dog","age":789,"_checked":false}]}
          </div>
        </form>
      </div>
    </div>
    <div
      class="ant-space-item"
    >
      <div
        class="ant-space ant-space-horizontal ant-space-align-center"
      >
        <div
          class="ant-space-item"
        >
          <button
            class="ant-btn ant-btn-primary"
            type="button"
          >
            <span>
              Toggle第二行的选中状态
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Hidden 1`] = `
<DocumentFragment>
  <div
    class="ant-formily-layout"
  >
    <form>
      <div
        class="ant-table-wrapper antd_formily_table_boost formily_antd_selection_display_none undefined"
      >
        <div
          class="ant-spin-nested-loading"
        >
          <div
            class="ant-spin-container"
          >
            <div
              class="ant-table"
            >
              <div
                class="ant-table-container"
              >
                <div
                  class="ant-table-content"
                >
                  <table
                    style="table-layout: auto;"
                  >
                    <colgroup>
                      <col
                        class="ant-table-selection-col"
                      />
                    </colgroup>
                    <thead
                      class="ant-table-thead"
                    >
                      <tr>
                        <th
                          class="ant-table-cell ant-table-selection-column"
                        />
                        <th
                          class="ant-table-cell"
                        >
                          名字
                        </th>
                        <th
                          class="ant-table-cell"
                        >
                          年龄
                        </th>
                      </tr>
                    </thead>
                    <tbody
                      class="ant-table-tbody"
                    >
                      <tr
                        class="ant-table-row ant-table-row-level-0"
                        data-row-key="0"
                      >
                        <td
                          class="ant-table-cell ant-table-selection-column"
                        >
                          <label
                            class="ant-radio-wrapper"
                          >
                            <span
                              class="ant-radio"
                            >
                              <input
                                class="ant-radio-input"
                                type="radio"
                                value=""
                              />
                              <span
                                class="ant-radio-inner"
                              />
                            </span>
                          </label>
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          fish
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          123
                        </td>
                      </tr>
                      <tr
                        class="ant-table-row ant-table-row-level-0"
                        data-row-key="1"
                      >
                        <td
                          class="ant-table-cell ant-table-selection-column"
                        >
                          <label
                            class="ant-radio-wrapper"
                          >
                            <span
                              class="ant-radio"
                            >
                              <input
                                class="ant-radio-input"
                                type="radio"
                                value=""
                              />
                              <span
                                class="ant-radio-inner"
                              />
                            </span>
                          </label>
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          cat
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          456
                        </td>
                      </tr>
                      <tr
                        class="ant-table-row ant-table-row-level-0"
                        data-row-key="2"
                      >
                        <td
                          class="ant-table-cell ant-table-selection-column"
                        >
                          <label
                            class="ant-radio-wrapper"
                          >
                            <span
                              class="ant-radio"
                            >
                              <input
                                class="ant-radio-input"
                                type="radio"
                                value=""
                              />
                              <span
                                class="ant-radio-inner"
                              />
                            </span>
                          </label>
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          dog
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          789
                        </td>
                      </tr>
                    </tbody>
                  </table>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div>
        {"data":[{"name":"fish","age":123,"_radio":false},{"name":"cat","age":456,"_radio":false},{"name":"dog","age":789,"_radio":false}]}
      </div>
    </form>
  </div>
</DocumentFragment>
`;

exports[`Radio 1`] = `
<DocumentFragment>
  <div
    class="ant-space ant-space-vertical"
    style="display: flex;"
  >
    <div
      class="ant-space-item"
      style="margin-bottom: 10px;"
    >
      <div
        class="ant-formily-layout"
      >
        <form>
          <div
            class="ant-table-wrapper antd_formily_table_boost undefined"
          >
            <div
              class="ant-spin-nested-loading"
            >
              <div
                class="ant-spin-container"
              >
                <div
                  class="ant-table"
                >
                  <div
                    class="ant-table-container"
                  >
                    <div
                      class="ant-table-content"
                    >
                      <table
                        style="table-layout: auto;"
                      >
                        <colgroup>
                          <col
                            class="ant-table-selection-col"
                          />
                        </colgroup>
                        <thead
                          class="ant-table-thead"
                        >
                          <tr>
                            <th
                              class="ant-table-cell ant-table-selection-column"
                            />
                            <th
                              class="ant-table-cell"
                            >
                              名字
                            </th>
                            <th
                              class="ant-table-cell"
                            >
                              年龄
                            </th>
                          </tr>
                        </thead>
                        <tbody
                          class="ant-table-tbody"
                        >
                          <tr
                            class="ant-table-row ant-table-row-level-0"
                            data-row-key="0"
                          >
                            <td
                              class="ant-table-cell ant-table-selection-column"
                            >
                              <label
                                class="ant-radio-wrapper"
                              >
                                <span
                                  class="ant-radio"
                                >
                                  <input
                                    class="ant-radio-input"
                                    type="radio"
                                    value=""
                                  />
                                  <span
                                    class="ant-radio-inner"
                                  />
                                </span>
                              </label>
                            </td>
                            <td
                              class="ant-table-cell"
                            >
                              fish
                            </td>
                            <td
                              class="ant-table-cell"
                            >
                              123
                            </td>
                          </tr>
                          <tr
                            class="ant-table-row ant-table-row-level-0"
                            data-row-key="1"
                          >
                            <td
                              class="ant-table-cell ant-table-selection-column"
                            >
                              <label
                                class="ant-radio-wrapper"
                              >
                                <span
                                  class="ant-radio"
                                >
                                  <input
                                    class="ant-radio-input"
                                    type="radio"
                                    value=""
                                  />
                                  <span
                                    class="ant-radio-inner"
                                  />
                                </span>
                              </label>
                            </td>
                            <td
                              class="ant-table-cell"
                            >
                              cat
                            </td>
                            <td
                              class="ant-table-cell"
                            >
                              456
                            </td>
                          </tr>
                          <tr
                            class="ant-table-row ant-table-row-level-0"
                            data-row-key="2"
                          >
                            <td
                              class="ant-table-cell ant-table-selection-column"
                            >
                              <label
                                class="ant-radio-wrapper"
                              >
                                <span
                                  class="ant-radio"
                                >
                                  <input
                                    class="ant-radio-input"
                                    type="radio"
                                    value=""
                                  />
                                  <span
                                    class="ant-radio-inner"
                                  />
                                </span>
                              </label>
                            </td>
                            <td
                              class="ant-table-cell"
                            >
                              dog
                            </td>
                            <td
                              class="ant-table-cell"
                            >
                              789
                            </td>
                          </tr>
                        </tbody>
                      </table>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div>
            {"data":[{"name":"fish","age":123,"_radio":false},{"name":"cat","age":456,"_radio":false},{"name":"dog","age":789,"_radio":false}]}
          </div>
        </form>
      </div>
    </div>
    <div
      class="ant-space-item"
    >
      <div
        class="ant-space ant-space-horizontal ant-space-align-center"
      >
        <div
          class="ant-space-item"
        >
          <button
            class="ant-btn ant-btn-primary"
            type="button"
          >
            <span>
              Toggle第二行的选中状态
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
